### PR TITLE
🌐 Add French translation for `docs/async.md`

### PR DESCRIPTION
**Pull request by Smlep**
Hey everyone!

Here is the PR to translate the async await doc page section into french.
See the french translation tracking issue [here](https://github.com/tiangolo/fastapi/issues/1972).

Thanks for the reviews👌

### CODE REVIEW

File: docs/fr/docs/async.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+# Concurrence et les mots-clés async et await
+
+Cette page vise à fournir des détails sur la syntaxe `async def` pour les *fonctions de chemins* et quelques rappels sur le code asynchrone, la concurrence et le parallélisme.
+
+## Vous êtes pressés ?
+
+<abbr title="'too long; didn't read' en anglais, ou 'trop long ; j'ai pas lu'"><strong>TL;DR :</strong></abbr>
+
+Si vous utilisez des fonctions ant d'autres bibliothèques qui nécessitent d'être appelées avec `await`, telles que :
```
Comment (rjNemo) [inline docs/fr/docs/async.md:9]:
```suggestion
Si vous utilisez des fonctions et d'autres bibliothèques qui nécessitent d'être appelées avec `await`, telles que :
```

File: docs/fr/docs/async.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+* une opération effectuée à distance par une API se termine
+* une opération en BDD se termine
+* une requête à une BDD renvoie un résultat
+* etc.
+
+Le temps d'exécution étant consommé majoritairement par l'attente d'opérations <abbr title="Input/Output ou Entrées et Sorties ">I/O</abbr> on appelle ceci des opérations <a href="https://fr.wikipedia.org/wiki/I/O_bound" class="external-link" target="_blank">"I/O bound"</a>.
+
+Ce concept se nomme l'"asynchronisme" car l'ordinateur / le programme n'a pas besoin d'être "synchronisé" avec la tâche, attendant le moment exact où cette dernière se terminera en ne faisant rien, pour être capable de récupérer le résultat de la tâche et l'utiliser dans la suite des opérations.
+
+À la place, en étant "asynchrone", une fois terminée, une tâche peut légèrement attendre (quelques microseconds) que l'ordinateur / le programme finisse ce qu'il était en train de faire, et revienne récupérer le résultat.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:88]:
```suggestion
À la place, en étant "asynchrone", une fois terminée, une tâche peut légèrement attendre (quelques microsecondes) que l'ordinateur / le programme finisse ce qu'il était en train de faire, et revienne récupérer le résultat.
```

File: docs/fr/docs/async.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+Ensuite, quand c'est votre tour, vous faites des actions "productives" 🤓, vous étudiez le menu, décidez ce que vous voulez, demandez à votre crush 😍 son choix, payez 💸, vérifiez que vous utilisez la bonne carte de crédit, vérifiez que le montant débité sur la carte est correct, vérifiez que la commande contient les bons produits, etc.
+
+Mais ensuite, même si vous n'avez pas encore vos burgers 🍔, votre travail avec le serveur 💁 est "en pause" ⏸, car vous devez attendre 🕙 que vos burgers soient prêts.
+
+Après vous être écarté du comptoir et vous être assis à votre table avec le numéro de votre commande, vous pouvez tourner 🔀 votre attention vers votre crush 😍, et "travailler" ⏯ 🤓 là-dessus. Vous êtes donc à nouveau en train de faire quelque chose de "productif" 🤓, vous flirtez avec votre crush 😍.
+
+Puis le seveur 💁 dit "J'ai fini de préparer les burgers" 🍔 en mettant votre numéro sur l'affichage du comptoir, mais vous ne courrez pas immédiatement au moment où votre numéro s'affiche. Vous savez que personne ne volera vos burgers 🍔 car vous avez votre numéro et les autres clients ont le leur.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:136]:
```suggestion
Puis le serveur 💁 dit "J'ai fini de préparer les burgers" 🍔 en mettant votre numéro sur l'affichage du comptoir, mais vous ne courrez pas immédiatement au moment où votre numéro s'affiche. Vous savez que personne ne volera vos burgers 🍔 car vous avez votre numéro et les autres clients ont le leur.
```

File: docs/fr/docs/async.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+Dans ce scénario de burgers parallèles, vous êtes un ordinateur / programme 🤖 avec deux processeurs (vous et votre crush 😍) attendant 🕙 à deux et dédiant votre attention 🕙 à "attendre devant le comptoir" pour une longue durée.
+
+Le fast-food a 8 processeurs (serveurs/cuisiniers) 👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳. Alors que le fast-food de burgers concurrents en avait 2 (un serveur et un cuisinier).
+
+Et pourtant l'expérience finale n'est pas meilleure 😞.
+
+---
+
+C'est donc l'histoire équivalente parrallèle pour les burgers 🍔.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:182]:
```suggestion
C'est donc l'histoire équivalente parallèle pour les burgers 🍔.
```

File: docs/fr/docs/async.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+
+---
+
+Il n'y a plus d'attente 🕙 nulle part, juste beaucoup de travail à effectuer, dans différentes pièces de la maison.
+
+Vous pourriez diviser en différentes sections comme avec les burgers, d'abord le salon, puis la cuisine, etc. Mais vous n'attendez 🕙 rien, vous ne faites que nettoyer et nettoyer, la séparation en sections ne changerait rien au final.
+
+Cela prendrait autant de temps pour finir avec ou sans sections (concurrence) et vous auriez effectué la même quantité de travail.
+
+Mais dans ce cas, si pouviez amener 8 ex-serveurs/cuisiners/devenus-nettoyeurs 👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳, et que chacun d'eux (plus vous) pouvait prendre une zone de la maison pour la nettoyer, vous pourriez faire tout le travail en parrallèle, et finir plus tôt.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:238]:
```suggestion
Mais dans ce cas, si pouviez amener 8 ex-serveurs/cuisiniers/devenus-nettoyeurs 👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳👨‍🍳, et que chacun d'eux (plus vous) pouvait prendre une zone de la maison pour la nettoyer, vous pourriez faire tout le travail en parallèle, et finir plus tôt.
```

File: docs/fr/docs/async.md [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+Mais ensuite, même si vous n'avez pas encore vos burgers 🍔, votre travail avec le serveur 💁 est "en pause" ⏸, car vous devez attendre 🕙 que vos burgers soient prêts.
+
+Après vous être écarté du comptoir et vous être assis à votre table avec le numéro de votre commande, vous pouvez tourner 🔀 votre attention vers votre crush 😍, et "travailler" ⏯ 🤓 là-dessus. Vous êtes donc à nouveau en train de faire quelque chose de "productif" 🤓, vous flirtez avec votre crush 😍.
+
+Puis le seveur 💁 dit "J'ai fini de préparer les burgers" 🍔 en mettant votre numéro sur l'affichage du comptoir, mais vous ne courrez pas immédiatement au moment où votre numéro s'affiche. Vous savez que personne ne volera vos burgers 🍔 car vous avez votre numéro et les autres clients ont le leur.
+
+Vous attendez donc que votre crush 😍 finisse son histoire, souriez gentillement et dites que vous allez chercher les burgers ⏸.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:138]:
```suggestion
Vous attendez donc que votre crush 😍 finisse son histoire, souriez gentiment et dites que vous allez chercher les burgers ⏸.
```

File: docs/fr/docs/async.md [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+Par exemple :
+
+* Traitements d'**audio** et d'**images**.
+* La **vision par ordinateur** : une image est composée de millions de pixels, chaque pixel ayant 3 valeurs / couleurs, les traiter tous va nécessiter d'effectuer des traitements sur chaque pixel, et de préférence tous en même temps.
+* L'apprentissage automatique (ou **Machine Learning**) : cela nécessite de nombreuses multiplications de matrices et vecteurs. Imaginez une énorme feuille de calcul remplie de nombres que vous multiplierez entre eux tous au même moment.
+* L'apprentissage profond (ou **Deep Learning**) : est un sous-domaine du **Machine Learning**, donc les mêmes raisons s'appliquent. Avec la différence qu'il n'y a pas une unique feuille de calcul de nombres à multiplier, mais une énorme quantité d'entre elles, et dans de nombreux cas, on utilise un processeur spécial pour construire et / ou utiliser ces modèles.
+
+### Concurrence + Parallélisme : Web + Machine Learning
+
+Avec **FastAPI** vous pouvez bénéficier de la concurrence mais vous pouvez aussi profiter du parallélisme et mutliprocessing afin de gérer des charges **CPU bound** qui sont récurrentes dans les systèmes de *Machine Learning*.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:257]:
```suggestion
Avec **FastAPI** vous pouvez bénéficier de la concurrence mais vous pouvez aussi profiter du parallélisme et multiprocessing afin de gérer des charges **CPU bound** qui sont récurrentes dans les systèmes de *Machine Learning*.
```

File: docs/fr/docs/async.md [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+
+Ça, ajouté au fait que Python soit le langage le plus populaire pour la **Data Science**, le **Machine Learning** et surtout le **Deep Leaning**, font de **FastAPI** un très bon choix pour les APIs de **Data Science** / **Machine Learning**.
```
Comment (rjNemo) [inline docs/fr/docs/async.md:259]:
```suggestion
Ça, ajouté au fait que Python soit le langage le plus populaire pour la **Data Science**, le **Machine Learning** et surtout le **Deep Learning**, font de **FastAPI** un très bon choix pour les APIs de **Data Science** / **Machine Learning**.
```

**Review by rjNemo — APPROVED** [review]

**Review by Cajuteq — CHANGES_REQUESTED** [review]
I left comments on the section that could be better imo. It is great work overall and the suggested changes may not all be relevant.

**Comment by Smlep** [comment]
@Cajuteq I've updated the PR according to your suggestions, if everything's fine with you now you can approve the PR.

**Review by Cajuteq — APPROVED** [review]

**Comment by tiangolo** [comment]
Awesome, thank you @Smlep! 🚀

And thank you for the reviews @Cajuteq and @rjNemo :bow:

**Pull request merged**